packages/next/src/server/app-render/action-utils.ts

```diff
@@ -18,10 +18,19 @@ export function createServerModuleMap({
     {},
     {
       get: (_, id: string) => {
-        const workerEntry =
+        const workers =
           serverActionsManifest[
             process.env.NEXT_RUNTIME === 'edge' ? 'edge' : 'node'
-          ][id].workers[normalizeWorkerPageName(pageName)]
+          ][id].workers
+        let workerEntry = workers[normalizeWorkerPageName(pageName)]
+
+        if (workerEntry === undefined) {
+          // Workaround due to faulty singleton.
+          for (let key in workers) {
+            workerEntry = workers[key]
+            break
+          }
+        }
```

Comment:
This workaround should not be used since it lets actions of other pages pass through without middleware checks.

It's needed because the singleton updates while building multiple pages in parallel.